### ISSUE TITLE
Mailchimp integration issues with fields in group

### ISSUE THREAD

**Issue by bavocorteville** [verbb/formie#697]
**Description**
I have a few issues with the Mailchimp integration:

1. The following error appears when you map the opt-in field to an agree field in a group.
```
[error][formie] Error when fetching opt-in-field: “Invalid field handle: newsletter” /app/cms/vendor/craftcms/cms/src/base/Element.php:4273
[info][formie] Mailchimp: Unable to find field “{newsletter}” for opt-in in submission.
```

2. The following error appears when you map the mailchimp mail to a mail field in a group.
```
[error][formie] Mailchimp: API error: “{"type":"https://mailchimp.com/developer/marketing/docs/errors/","title":"Invalid Resource","status":400,"detail":"The resource submitted could not be validated. For field-specific details, see the 'errors' array.","instance":"4f09ebb5-a639-ef29-96d9-be4fbf56a711","errors":[{"field":"email_address","message":"This value should not be blank."}]}” /app/cms/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113
```

Both are working correctly when I remove those fields from the group and re-add them outside a group.

3. A name field cannot be split into first name and last name for mapping the name fields if the fields are in a group.

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: No
- Custom Form Templates: Yes

**Additional info**
- Plugin version: 1.5.3
- Craft version: 3.7.27.1
- Multi-site: Yes

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.3",
  "...": "..."
}
```

Then run `composer update`. Please note that you'll need to re-map the "Opt-in" field after this update, as the way we were storing the field handle for fields within a Group field was incorrect.

**Comment by bavocorteville** [comment]
@engram-design Thanks! I can't test it at the moment because I get this error when I go to a form (/admin/formie/forms/edit/{id}):

`Return value of craft\models\FieldLayout::getTabs() must be of the type array, null returned`

EDIT: `composer update` also updated Craft 3.7.27.1 to 3.7.28. It works in the old version, but not in the new version of Craft.
EDIT2: I tested your fix in 3.7.27.1. The first issue is fixed, the other two still exist.

**Comment by engram-design** [comment]
Fixed in [1.5.4](https://github.com/verbb/formie/releases/tag/1.5.4)

**Comment by bavocorteville** [comment]
@engram-design These two issues have not yet been resolved.

1. The following error appears when you map the mailchimp mail to a mail field in a group.
`[error][formie] Mailchimp: API error: “{"type":"https://mailchimp.com/developer/marketing/docs/errors/","title":"Invalid Resource","status":400,"detail":"The resource submitted could not be validated. For field-specific details, see the 'errors' array.","instance":"4f09ebb5-a639-ef29-96d9-be4fbf56a711","errors":[{"field":"email_address","message":"This value should not be blank."}]}” /app/cms/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113`
2. A name field cannot be split into first name and last name for mapping the name fields if the fields are in a group.

**Comment by engram-design** [comment]
@bavodeclercq Sorry I did actually miss that, likely in your edits. It wasn't possible to be able to map to sub-fields, when they existed inside a nesting field (Name in Group for instance), but I've refactored and re-organised things to make that possible).

To get these fixes early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.4",
  "...": "..."
}
```

Then run `composer update`.
